### PR TITLE
Group scoring functions

### PR DESCRIPTION
**Pull request by giovannidoni**
In this PR, I am moving things around to separate function according to their function. This makes the code a bit easier to read for future refactoring.

### CODE REVIEW

File: swarmcg/utils.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import contextlib
+import random
+import sys
+
+import MDAnalysis as mda
+
+
```
Comment (CharlyEmpereurmot) [inline swarmcg/utils.py:7]: The reason why I was using this, is that FST-PSO has his own output that I don't want to show to the users (we still acknowledge them correctly in the header displayed at program execution). We do the necessary checks for input consistency + progress/metrics informations ourselves, so we really don't care about FST-PSO error messages or progress informations. I'm not sure this is the correct way to handle the IO for this kind of things though.

File: swarmcg/simulations/runner.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            _ = gmx_process.communicate()[1].decode()
+            gmx_process.kill()
+
+        if gmx_process.returncode == 0:
+            return self
+        else:
+            msg = (
+                f"Gromacs grompp failed at MD {self.step_name} step, see its error message above. " 
+                f"You may also want to check the parameters of the MDP file provided through "
+                f"argument -cg_sim_mdp_mini. If you think this is a bug, please consider opening "
```
Comment (CharlyEmpereurmot) [inline swarmcg/simulations/runner.py:97]: Here the message may refer to arguments -cg_sim_mdp_mini if the error happened at the minimization step, but it has to be adapted and refer to the correct arguments for also the equilibration and md/production steps.

File: swarmcg/scoring/bonds.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import MDAnalysis as mda
+import numpy as np
+
+from swarmcg import config
+
+
```
Comment (CharlyEmpereurmot) [inline swarmcg/scoring/bonds.py:6]: These 2 functions can be merged easily into a single one, I will do this soon. Same thing for angles and dihedrals.

**Review by CharlyEmpereurmot — COMMENTED** [review]
Already looks better :+1: I will run the tests I currently have in my PyCharm on the branch before merging.

File: swarmcg/simulations/runner.py (hunk 2 above)
Comment (giovannidoni) [reply swarmcg/simulations/runner.py:97]: I missed that, I can add another argument to the class for this or via subclassing

File: swarmcg/utils.py (hunk 1 above)
Comment (giovannidoni) [reply swarmcg/utils.py:7]: will figure it out, I guess

**Review by CharlyEmpereurmot — APPROVED** [review]

**Pull request merged**